src/main/java/com/pusher/client/Pusher.java

```diff
@@ -191,12 +191,12 @@ public void connect(final ConnectionEventListener eventListener, ConnectionState
      * Disconnect from Pusher.
      *
      * <p>
-     * Calls are ignored if the {@link Connection#getState()}, retrieved from {@link Pusher#getConnection}, is not
-     * {@link com.pusher.client.connection.ConnectionState#CONNECTED}.
+     * Calls are ignored if the {@link Connection#getState()}, retrieved from {@link Pusher#getConnection}, is
+     * {@link com.pusher.client.connection.ConnectionState#DISCONNECTING} or  {@link com.pusher.client.connection.ConnectionState#DISCONNECTED}.
      * </p>
      */
     public void disconnect() {
-        if (connection.getState() == ConnectionState.CONNECTED) {
+        if (connection.getState() != ConnectionState.DISCONNECTING && connection.getState() != ConnectionState.DISCONNECTED) {
```

Comment:
WDYT about having this rule encapsulated on `connection` instance?

Something like:
```java
if (connection.canBeDisconnected()) { .. }
```
This way we avoid repeating the same pattern elsewhere.

Reply:
Should we do the equivalent for the condition in the `connect` method? How would you call it?